p2p/peer.py

```diff
@@ -856,7 +854,9 @@ def _add_peer(self,
     async def _run(self) -> None:
         # FIXME: PeerPool should probably no longer be a BaseService, but for now we're keeping it
         # so in order to ensure we cancel all peers when we terminate.
-        self.run_task(self._periodically_report_stats())
+        if self.event_bus is not None:
+            self.run_daemon_task(self.handle_peer_count_requests())
```

Comment:
@cburgdorf can you :+1: / :-1: this change.  It moves the task into the `_run` method so that it only starts when the peer pool has started as well as making it a deamon task which will crash the whole trinity process if it fails.

Reply:
Yes, that makes sense. :+1:  That said, I think this will move soon into its own child service or something similar.